README.md

```diff
@@ -54,134 +55,62 @@ type Decorator = (value: Input, context: {
   isPrivate?: boolean;
   isStatic?: boolean;
   addInitializer?(initializer: () => void): void;
-+ metadataKey?: MetadataKey;
-+ class?: {
-+   metadataKey: MetadataKey;
-+   name: string;
-+ }
++ metadata?: Record<string | number | symbol, unknown>;
```

Comment:
In general I'm fine with this. The separate `class` property was intended to make it easier to distinguish between metadata assigned to a class, and metadata assigned to a function. In the event that _function decorators_ is adopted in the future, a decorator that is reusable on both functions and methods might want to be able to attach metadata to the function itself, so we may want to be somewhat forward-thinking about this so as not to paint ourselves into a corner.

Reply:
I don't think this really paints us into a corner, in the case of function decorators the metadata would be placed on the function directly via `Symbol.metadata`, and would otherwise have the same API. The main difference would be that method decorators will need to address the fact that they share metadata with the rest of the class, but the `kind` property of context should allow decorators to handle both cases pretty easily.

I prefer this because it's both simpler to implement and to reason about, and it's more consistent across all types of class elements. If we were to place function metadata on the method, that could get tricky fast. It means for getters and setters you need to read the PropertyDescriptor to get the metadata, and for accessors it's not clear which element the metadata would go on. This resolves those issues and provides a clearer mental model.